demo-circuit/src/sc2sc/tests.rs

```diff
@@ -0,0 +1,565 @@
+            FieldElement::zero(),
+            FieldElement::zero(),
+            next_cert,
+            curr_cert,
+            ScCommitmentCertPath::default(),
+            ScCommitmentCertPath::default(),
+            GingerMHTBinaryPath::new(vec![(FieldElement::default(), false); MSG_MT_HEIGHT]),
+        );
+    }
+
+    #[test]
+    #[should_panic(expected = "least 3 custom fields")]
+    fn certificates_have_less_than_3_custom_fields() {
+        let curr_cert = RandomWithdrawalCertificateDataBuilder::new(2).build();
+        let next_cert = RandomWithdrawalCertificateDataBuilder::new(2).build();
+
+        Sc2Sc::new(
+            FieldElement::zero(),
+            FieldElement::zero(),
+            FieldElement::zero(),
+            next_cert,
+            curr_cert,
+            ScCommitmentCertPath::default(),
+            ScCommitmentCertPath::default(),
+            GingerMHTBinaryPath::new(vec![(FieldElement::default(), false); MSG_MT_HEIGHT]),
+        );
+    }
+}
+
+mod should_fail {
+    use super::*;
+
+    #[rstest]
+    #[serial]
+    #[should_panic(expected = "curr_cert.epoch + 1 == next_cert.epoch")]
+    fn if_not_contiguos_epochs(
+        mut rng: ThreadRng,
+        base_commitments: CommitmentPair,
+        #[values(true, false)] zk: bool,
+    ) {
+        let sc_id: FieldElement = rng.gen();
+        // Just one message in the root
+        let (msg_root, msg_hash, msg_path) = test_utils::messages(&mut rng, 1, 0);
+
+        let (curr, next) = base_commitments;
+
+        let mut curr = curr
+            .with_certificate_msg_root(msg_root)
+            .generate_sc_data(None, &mut rng, sc_id);
+        let (curr_cert, curr_cert_path, curr_sc_tx_commitment) =
+            curr.get_widthdrawal_certificate_info(0);
+
+        let mut next = next
+            .with_max_quality_certificate_hash(curr_cert.hash().unwrap())
+            .with_epoch(1234) // Change next epoch
+            .generate_sc_data(None, &mut rng, sc_id);
+        let (next_cert, next_cert_path, next_sc_tx_commitment) =
+            next.get_widthdrawal_certificate_info(0);
+
+        let sc2sc = Sc2Sc::new(
+            next_sc_tx_commitment,
+            curr_sc_tx_commitment,
+            msg_hash,
+            next_cert,
+            curr_cert,
+            next_cert_path,
+            curr_cert_path,
+            msg_path,
+        );
+
+        assert_circuit(sc2sc, if zk { Some(&mut rng) } else { None });
+    }
+
+    enum TestChangeTxPathAction {
+        ScTxCommitmentRoot,
+        FTRootHash,
+        BTRootHash,
+        WCertRoot,
+        SSC,
+        CertPath,
+        ScCommitmentPath,
+    }
+
+    #[rstest]
+    #[serial]
+    #[should_panic(
+        expected = "(curr_sc_tx_commitment, curr_sc_tx_commitments_path) == curr_sc_tx_commitments_root"
+    )]
+    #[case::sc_tx_commitment_root(TestChangeTxPathAction::ScTxCommitmentRoot)]
+    #[should_panic(expected = "Current epoch sc_tx_commitment_root recostruction")]
```

Comment:
This won't be expected anymore once you peel off that extra constraint

Reply:
Right... I've removed the test case